src/language/templates/code_generator.cmake

```diff
@@ -0,0 +1,37 @@
+#
+# THIS FILE IS GENERATED AND SHALL NOT BE EDITED.
+#
+
+# cmake-format: off
+set(CODE_GENERATOR_JINJA_FILES
```

Comment:
I'm just curious, so it looks like you generate this cmake file too. How does that go into the build process. Don't you need all cmake files to be there already before you can run `cmake ..`? It's really cool that this works, but I don't quite understand how

Reply:
I have the same question. As far as I understood this file is added to the repo so that the `cmake` can run for the first time. It's not very clear to me how when we add an `ast` class then `cmake` is going to use the updated `code_generator.cmake` to set all the variables though

Reply:
OK, so this was explained elsewhere. The file is assumed to exist and is updated whenever a new template file is added. I'm not too happy that this file is generated, I would rather this step is done manually and then this cyclic dependency and possible confusion isn't there in the first place. But I also don't feel strongly about it either way.